### ISSUE TITLE
Concept for tools / keybindings triggering the same action

### ISSUE THREAD

**Issue by michael** [substance/substance#39]
Would like to look into this tomorrow. It's related to #23. Would be nice to have some generic concept for keybindings and somehow make this work nicely in combination with tools. E.g. cmd+b and clicking the strong tool should trigger the same action.

One thing we discussed was breaking out the AnnotationTool implementation into a ToggleAnnotation transformation. This must provide static functions for canCreate(sel), canExpand etc. that the tool implementation can use to update the button states. In performAction we just execute the transformation using a surface transaction. Parameter being the selection and the annotationType to be toggled.

With that interface we could also introduce custom keybindings that call the same function. Something like this:

``` js
var toggleAnnotation = require('substance/document/transformations/toggle_annotation');

var ToggleStrong = KeyBinding.extend({
  handles: ['command+b', 'ctrl+b'],
  performAction: function() {
    var surface = this.context.surfaceManager.getSurface();
    surface.transaction(function(tx, args) {
      args.annotationType = 'strong';
      toggleAnnotation(tx, args);
    });
  }
});
```

What do you think? We will need some orchestration to announce the desired keybindings to the app. For the Writer/Editor components this could just be passed like this:

``` js
var MyEditor = Editor.extend({
  config: {
    article: Article,
    toolbar: Toolbar,
    keybindings: [ToggleStrong, ToggleLink],
    components: {
      "paragraph": require('substance/ui/nodes/paragraph_component'),
      "link": require('substance/ui/nodes/link_component')
    }
  }
});
```

Could we still use the Substance Keyboarder to manage things? Need your input here generally before I implement this.

**Comment by obuchtala** [comment]
The registration should not be too difficult.
Don't know if the old Substance.Keyboard is really necessary. For sure you need the part of it where the commands get parsed. And maybe we end up reviving the Substance.Keyboard, yes.

**Comment by michael** [comment]
Fixed via #43
